### PR TITLE
Fix of Import failure - Cleanup folder created during Import

### PR DESCRIPTION
**Pull request by MeiravDaniel**
### What does this PR do?

This pull request replaces this pull request - 
https://github.com/codenvy/che-core/pull/706
It add a check if the root folder of the import exist before import
If it does and the import failed - it deletes the folder
### What issues does this PR fix or reference?

The problem of having a "leftover" folder when import failed and there was no folder before
### Previous Behavior

The folder was created and remain even if the import failed
### New Behavior

the folder is deleted on failure
### Tests written?

No
### Docs requirements?

No

Please review [Che's Contributing Guide](https://github.com/eclipse/che/CONTRIBUTING.MD) for best practices.

https://github.com/codenvy/che-core/pull/706/files
I would like to mention that the comment on "newDto" and Collection Map
was not fixed because the origin change did not change the body of the method
It only wrap them by try {} catch , so if a folder was created during the
Import - It will be deleted on failure

Signed-off-by: i028422 meirav.daniel@sap.com

### CODE REVIEW

**Comment by MeiravDaniel** [comment]
@tareqhs could you please review?
Thanks,
Meirav

**Review by tareksha — COMMENTED** [review]
Does it pass all unit tests?

**Comment by MeiravDaniel** [comment]
Yes.It does

File: platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java [hunk 1]
```diff
@@ -806,15 +806,25 @@ public ImportResponse importProject(@ApiParam(value = "Workspace ID", required =
...
         VirtualFileEntry virtualFile = getVirtualFile(workspace, path, force);
 
-        final FolderEntry baseProjectFolder = (FolderEntry)virtualFile;
-        importer.importSources(baseProjectFolder, projectSource.getLocation(), projectSource.getParameters(), outputOutputConsumerFactory);
-
-        //project source already imported going to configure project
-        return configureProject(importProject, baseProjectFolder, workspace, creationDate);
+        FolderEntry baseProjectFolder = null;
+        try {
+        	 baseProjectFolder = (FolderEntry)virtualFile;
```
Comment (skabashnyuk) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java:814]: wrong formatting

**Review by skabashnyuk — APPROVED** [review]

**Comment by skabashnyuk** [comment]
@tareqhs  can you confirm that formation is fixed?

**Pull request merged**